src/iterator.cr

```diff
@@ -131,13 +131,19 @@ module Iterator(T)
   end
 
   def self.of(&block : -> T)
-    SingletonProc(T).new(block)
+    SingletonProc(typeof(without_stop(&block))).new(block)
+  end
+
+  private def self.without_stop(&block : -> T)
+    e = block.call
+    raise "" if e.is_a?(Iterator::Stop)
```

Comment:
Oh, I see. Still, `raise ""` (which internally translates to `raise ArgumentError.new`, no?) looks odd for me...